### PR TITLE
snapshot refactor 2: move v2 api to super module

### PR DESCRIPTION
**Pull request by choleraehyq**
@siddontang @zhangjinpeng1987 PTAL

### CODE REVIEW

File: src/raftstore/store/snap.rs [hunk 1]
```diff
@@ -165,8 +170,8 @@ pub trait Snapshot: Read + Write + Send {
 // Only used in tests.
 pub fn copy_snapshot(mut from: Box<Snapshot>, mut to: Box<Snapshot>) -> io::Result<()> {
     if !to.exists() {
-        try!(io::copy(&mut from, &mut to));
-        try!(to.save());
+        io::copy(&mut from, &mut to)?;
```
Comment (siddontang) [inline src/raftstore/store/snap.rs]: don't use ? now.

Comment (andelf) [inline src/raftstore/store/snap.rs]:
```rust
src/bin/tikv-ctl.rs:426:            let (region_id, suffix) = keys::decode_region_meta_key(key)?;
src/raftstore/store/snap.rs:999:                    let handle = snap.cf_handle(cf_file.cf)?;
```

Comment (choleraehyq) [inline src/raftstore/store/snap.rs]: https://github.com/pingcap/tikv/blob/pre-ga/tests/raftstore/node.rs#L106

Comment (choleraehyq) [inline src/raftstore/store/snap.rs]: Seems all `?` were written by me. But no one told me this before. Because of performance issue?

Comment (zhangjinpeng1987) [inline src/raftstore/store/snap.rs]: style

**Comment by zhangjinpeng1987** [comment]
rest LGTM

File: src/raftstore/store/snap.rs (hunk 1 above)
Comment (siddontang) [inline src/raftstore/store/snap.rs]: I see that you have already introduced `?`, please file a PR to remove all later.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by choleraehyq** [comment]
@hhkbp2 @siddontang @BusyJay ptal

File: src/raftstore/store/store.rs [hunk 2]
```diff
@@ -1703,11 +1703,10 @@ impl<T: Transport, C: PdClient> Store<T, C> {
     }
 
     fn handle_snap_mgr_gc(&mut self) -> Result<()> {
-        let mut snap_keys = try!(self.snap_mgr.list_idle_snap());
+        let snap_keys = try!(self.snap_mgr.list_idle_snap());
         if snap_keys.is_empty() {
             return Ok(());
         }
-        snap_keys.sort();
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: why remove sort here?

Comment (choleraehyq) [inline src/raftstore/store/store.rs]: in list_idle_snap keys are already sorted.

Comment (choleraehyq) [inline src/raftstore/store/store.rs]: https://github.com/pingcap/tikv/blob/pre-ga/src/raftstore/store/snap.rs#L1625

**Comment by siddontang** [comment]
LGTM

**Pull request merged**